### DISCUSSION TITLE
Can you add the function of minimizing windows to the taskbar?

### DISCUSSION THREAD

**Discussion by HeliusHao** [massCodeIO/massCode#114] [Q&A]
Can you add the function of minimizing windows to the taskbar? thanks

**Comment by alexn-s** [comment]
great idea.
+1

**Comment by antonreshetov** [comment]
@HeliusHao I'm not quite sure what you mean exactly. The application can be minimized simply by clicking on the minimize button in the window control section.

**Reply by HeliusHao** [reply]
Minimize to tray

![image](https://user-images.githubusercontent.com/33107305/178200966-a9117749-0800-48b7-99a2-4d455564ca62.png)

**Reply by sangyuxiaowu** [reply]
> @HeliusHao I'm not quite sure what you mean exactly. The application can be minimized simply by clicking on the minimize button in the window control section.

Will this feature be added? I probably researched, this seems to support not only windows, but also mac and some linux.
I tried adding it, but I found that there will be some problems with exiting. Need to redesign the exit function.

https://www.electronjs.org/docs/latest/tutorial/tray

**Reply by antonreshetov** [reply]
I don't plan to add this feature yet.

**Reply by sangyuxiaowu** [reply]
> I don't plan to add this feature yet.

Would you mind if I try it? 😀

**Reply by antonreshetov** [reply]
> > I don't plan to add this feature yet.
> 
> Would you mind if I try it? 😀

Not yet needed.

**Reply by sangyuxiaowu** [reply]
> 



> > > I don't plan to add this feature yet.
> > 
> > 
> > Would you mind if I try it? 😀
> 
> Not yet needed.

ok, mainly because I often close it by mistake.

**Reply by antonreshetov** [reply]
The "X" button in macOS is intended initially to hide, while the "X" in Windows, as I recall, is a direct action to close the window, i.e. to exit. So it violates the native use of the window.

**Reply by sangyuxiaowu** [reply]
Well, I understand what you mean.

In fact, there is no conflict. There is no problem with the "X" button closing the form, but the APP also exits. Other plugins don't work either.

Plus the taskbar tray, so the program can always run, waiting for the call at any time.

The main problem here is that users in different places have different habits.

Maybe add a setting?

![image](https://user-images.githubusercontent.com/17669599/180718443-df203d02-a5f7-4e65-b573-1b7c598d8f6f.png)

**Reply by onetrev** [reply]
> The "X" button in macOS is intended initially to hide, while the "X" in Windows, as I recall, is a direct action to close the window, i.e. to exit. So it violates the native use of the window.

Hey @antonreshetov ... You are right in Windows this isn't default / native fuctionality, but there are numerous Windows apps that offer this, especially things like chat apps (Skype, Signal, etc.) where you want the program running all the time, but don't want to see it in your task bar all day. Because it isn't the native functionality Windows apps that do offer minimize to tray have a setting / option so people can enable the "X" to minimize to tray instead of closing the app.

For example, Signal added this functionality a little while ago (ex: https://github.com/signalapp/Signal-Desktop/commit/af1f2ea44927d92327acda3f20503e7715d4dd89).

**Comment by SamKr** [comment]
+1, since this app takes a long time to load, it'd be great to have it sitting in the system tray for instant pullup (and not take taskbar space)

**Comment by aderchox** [comment]
Here's a temporary solution (Sorry if it's a bit long, but I promise it makes your life easy in the end) for those Windows users who might need this feature so badly (just like myself 😄): First get [RBTray](https://github.com/benbuck/rbtray/releases), then press `Win+R` and type `shell:startup`, in the folder that opens, put two shortcuts to both RBTray and massCode. Now install [StrokesPlus](https://www.strokesplus.net/Downloads) and open it. Now assign the "Down" gesture (or whatever else gesture you prefer yourself personally) to the hotkey `Ctrl + Alt + Down`. This hotkey is the hotkey that triggers RBTray to put an app into the tray icons section.
Done! Now hold the right click of the mouse and pull it down in a straight line (or whatever else gesture you chose), and massCode will go into the tray icons section!
![image](https://github.com/massCodeIO/massCode/assets/52244768/c3e722fd-313b-471c-a52c-beaea00902ea)
Also a word with the great author: Although this feature violates the native functionality of Windows OS, it can at least be an optional setting, and even famous apps (eg, Discord) do it out of the box. Thank you for your amazing work anyway.

**Comment by onetrev** [comment]
I believe this is now solved with the latest release, v3.12.0, and this commit in particular: https://github.com/massCodeIO/massCode/pull/499

**Reply by aderchox** [reply]
To the taskbar yes, to the tray no. (I also don't see how adding a button for minimizing to taskbar could be useful when the native Window has a "—" button for minimization, at the top right corner next to the expansion and closing buttons. But anyways, it's there now. But the more useful feature would have been minimization to tray, to remove the need for hacks like RBTray.)

**Comment by JeremyDufeux** [comment]
+1 Minimize to tray would be a great feature
